src/api/db/connect.ts

```diff
@@ -1,34 +1,32 @@
 import mongoose from 'mongoose';
 import { InternalServerError } from '../errors.js';
+import { Config } from '../../config/config.js';
 
 // TODO: consider using multiple connections (one per model)
 // to reduce risk of slow trains
 // https://mongoosejs.com/docs/connections.html#multiple_connections
 // https://thecodebarbarian.com/slow-trains-in-mongodb-and-nodejs
 
-let cachedConnectionPromise = null;
+let client: typeof mongoose | null = null;
```

Comment:
2 bits of confusion on this line:

1. is it ok practice to just use `typeof <some_imported_node_module`? I would have guessed that you have to explicitly import and use the Mongoose types.
2. I think this `mongoose.connect()` can also return a Promise

Reply:
but I suppose if we're awaiting the promise it would have already resolved by the time it gets assigned to `client`... maybe answered my own question here?